src/main/java/io/usethesource/vallang/io/binary/util/DirectZstdInputStream.java

```diff
@@ -28,7 +28,11 @@ protected ByteBuffer refill(ByteBuffer torefill) throws IOException {
 
     private static ByteBuffer constructDecompressedBuffer(ByteBufferInputStream oriStream) {
         int compressedSize = oriStream.getByteBuffer().remaining();
-        ByteBuffer result = DirectByteBufferCache.getInstance().get(Math.min(compressedSize * 2, ZstdDirectBufferDecompressingStream.recommendedTargetBufferSize()));
+        int bufferSize = ZstdDirectBufferDecompressingStream.recommendedTargetBufferSize();
+        if (bufferSize > compressedSize) {
+            bufferSize = Math.min(compressedSize * 2, bufferSize);
```

Comment:
@DavyLandman why wouldn't this cause an 32-bit-int overflow sometimes? is the compressedSize limited to half of INT_MAX? Because the overflow would make the Math.min produce a negative number.

Reply:
Because the guard before it, it only happens if the buffer size that zstd suggested (which is 128kb) is highter than the full compressed size. So than we know we can safely take a rough estimate if 2* the compressed size.

This is to avoid allocating big blocks for relatively small files.

Reply:
so `recommendedTargetBufferSize` never returns a value larger than or equal to MAX_INT / 2?